runelite-client/src/main/java/net/runelite/client/plugins/inventoryviewer/InventoryViewerOverlay.java

```diff
@@ -27,78 +28,196 @@
 
-	private final PanelComponent panelComponent = new PanelComponent();
+	private final PanelComponent wrapperComponent = new PanelComponent();
+	private final PanelComponent inventoryComponent = new PanelComponent();
+	private final TitleComponent freeSlotsComponent = TitleComponent.builder().build();
+
+	@Setter(AccessLevel.PACKAGE)
+	private Map<Integer, Integer> groupedItems = null;
+
+	@Setter(AccessLevel.PACKAGE)
+	private int remainingSpaces = -1;
+
+	private ImageComponent inventoryIconSprite;
 
 	@Inject
-	private InventoryViewerOverlay(Client client, ItemManager itemManager)
+	private InventoryViewerOverlay(Client client, ItemManager itemManager, SpriteManager spriteManager, InventoryViewerConfig config)
 	{
 		setPosition(OverlayPosition.BOTTOM_RIGHT);
-		panelComponent.setWrapping(4);
-		panelComponent.setGap(new Point(6, 4));
-		panelComponent.setOrientation(ComponentOrientation.HORIZONTAL);
+		inventoryComponent.setWrapping(INVENTORY_WIDTH);
+		inventoryComponent.setGap(SPRITE_PADDING);
+		inventoryComponent.setOrientation(ComponentOrientation.HORIZONTAL);
+		inventoryComponent.setBackgroundColor(null);
+		inventoryComponent.setBorder(new Rectangle(
+			ComponentConstants.STANDARD_BORDER,
+			0,
+			0,
+			ComponentConstants.STANDARD_BORDER));
+
+		wrapperComponent.setOrientation(ComponentOrientation.VERTICAL);
+		wrapperComponent.setWrapping(2);
+		wrapperComponent.setBorder(new Rectangle(
+			ComponentConstants.STANDARD_BORDER,
+			ComponentConstants.STANDARD_BORDER,
+			ComponentConstants.STANDARD_BORDER,
+			ComponentConstants.STANDARD_BORDER));
+
 		this.itemManager = itemManager;
 		this.client = client;
+		this.config = config;
+
+		spriteManager.getSpriteAsync(SpriteID.TAB_INVENTORY, 0, sprite ->
+		{
+			final BufferedImage img = new BufferedImage(Constants.ITEM_SPRITE_WIDTH, Constants.ITEM_SPRITE_HEIGHT, BufferedImage.TYPE_INT_ARGB);
+			final int centeredY = Constants.ITEM_SPRITE_HEIGHT / 2 - sprite.getHeight() / 2;
+
+			final Graphics2D g = img.createGraphics();
+			g.drawImage(sprite, 0, centeredY, null);
+			g.dispose();
+
+			inventoryIconSprite = new ImageComponent(img);
+		});
 	}
 
 	@Override
 	public Dimension render(Graphics2D graphics)
 	{
 		final ItemContainer itemContainer = client.getItemContainer(InventoryID.INVENTORY);
 
+		final Item[] items;
 		if (itemContainer == null)
 		{
-			return null;
+			items = new Item[0];
+		}
+		else
+		{
+			items = itemContainer.getItems();
 		}
 
-		panelComponent.getChildren().clear();
+		inventoryComponent.getChildren().clear();
+		wrapperComponent.getChildren().clear();
+		int wrapperWidth;
 
-		final Item[] items = itemContainer.getItems();
+		if (config.groupItems())
+		{
+			if (groupedItems == null)
+			{
+				// Store in a LinkedHashMap to preserve inventory ordering
+				groupedItems = Arrays.stream(items)
+					.filter(p -> p.getId() != -1)
+					.collect(groupingBy(Item::getId, LinkedHashMap::new, summingInt(Item::getQuantity)));
```

Comment:
Does this order the grouped items in the same order they appear in your inventory? From the screenshots provided on this PR, it seems like it does not.